### PR TITLE
Feature/mongo integration

### PR DESCRIPTION
**Pull request by bobrowadam**
@shaharke adding mongo-integration. Still need to add tests.

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  });
+});
+
+// Get - uses github.client to retrieve github status and sending result to the client.
+app.get('/', function(req, res) {
+  gitClient.getGitStatus().then(function(body) {
+    console.log('Got response from gitStat:\n\n\n #####################');
+    const promiseArray = [];
+    body.forEach((statusObject) => {
+      if (statusObject.length === undefined) {
```
Comment (shaharke) [inline src/app.js]: what are you trying to check, that `statusObject` is not an array?

File: src/status.dao.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const promise = require('bluebird');
+
+function StatusDAO(conn) {
+  let db = conn;
+  return {
+    writeStatus: (statusObject) => {
```
Comment (shaharke) [inline src/status.dao.js]: Change this to `insert`

File: src/status.dao.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      return new promise((fulfill, reject) => {
+        db.collection('statuses').find(statusObject).next((err, doc) => {
```
Comment (shaharke) [inline src/status.dao.js]: uggg... risky operation you have here. Answer this: what happens if your collection now contain 1 million documents?

File: src/status.dao.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+const promise = require('bluebird');
```
Comment (shaharke) [inline src/status.dao.js]: Add an `insertMany` function that uses Bulk Operations: https://mongodb.github.io/node-mongodb-native/api-generated/unordered.html

File: src/app.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        promiseArray.push(statusDao.writeStatus(statusObject));
```
Comment (shaharke) [inline src/app.js]: Once you've added `insertMany` to the DAO (see comment below), use that function instead

File: src/github.client.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const request = require('request');
+const promise = require("bluebird");
+const requestGitStatuses = require('./get.git.status');
+// const request = promise.promisify('request');
+
+function getGitStatus() {
+  return new Promise(function(fulfill, reject) {
+    requestGitStatuses().then(function(body) {
```
Comment (shaharke) [inline src/github.client.js]: Why did we extract this functionality to another module?

File: src/app.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
+const gitClient = require("./github.client.js");
+const LISTEN_PORT = 3222;
+const express = require('express');
+const app = express();
+const mongoSetup = require('./mongo.setup');
+const statusDaoFactory = require('./status.dao');
+const promise = require('bluebird');
+
+let statusDao = null;
```
Comment (shaharke) [inline src/app.js]: Nice work on the initialization part. Sucks to do dependency injection on your own huh?

**Comment by shaharke** [comment]
Once you've finished with adding `insertMany` to the DAO and using Bulk Operations, we can move to the next tasks: configuration files.

File: src/app.js (hunk 1 above)
Comment (bobrowadam) [inline src/app.js]: Yes.  'api/messages.json' sometimes returns '[]' and then then the body will look like this:
[[],{statusObject}]  

('api/last-message.json' returns 'statusObject'  )

Comment (shaharke) [inline src/app.js]: You need to solve it in the GitClient. You need to make sure you get a uniform and deterministic result from the client. No additional hacks that are being defensice.

File: src/status.dao.js (hunk 3 above)
Comment (bobrowadam) [inline src/status.dao.js]: Was thinking on 2 options:
1- search for a single unique field like Date.
Or
2 - Verify that the DB is Sorted by Date in every insert operation and then just compare the last documents Date field with the current status I want to insert.

Comment (shaharke) [inline src/status.dao.js]: Ok... those are good ideas but not solutions. How would you implement that with mongo?

**Comment by bobrowadam** [comment]
@shaharke take a look on my last commit. I Added some new stuff per your comments.

**Pull request closed without merging**